packages/runtimes/js/src/JSRuntime.js

```diff
@@ -632,39 +624,42 @@ function getRegisterCode(
   entryBundle: NamedBundle,
   bundleGraph: BundleGraph<NamedBundle>,
 ): string {
-  let idToName = {};
+  let idToName = [];
   bundleGraph.traverseBundles((bundle, _, actions) => {
     if (bundle.bundleBehavior === 'inline') {
       return;
     }
 
-    idToName[bundle.publicId] = path.basename(nullthrows(bundle.name));
+    idToName.push([
+      bundle.publicId,
+      relativeBundlePath(entryBundle, nullthrows(bundle), {
+        leadingDotSlash: false,
+      }),
+    ]);
 
     if (bundle !== entryBundle && isNewContext(bundle, bundleGraph)) {
       for (let referenced of bundleGraph.getReferencedBundles(bundle)) {
-        idToName[referenced.publicId] = path.basename(
-          nullthrows(referenced.name),
-        );
+        idToName.push([
+          referenced.publicId,
+          relativeBundlePath(entryBundle, nullthrows(referenced), {
+            leadingDotSlash: false,
+          }),
+        ]);
       }
       // New contexts have their own manifests, so there's no need to continue.
       actions.skipChildren();
     }
   }, entryBundle);
 
-  return (
-    "require('./helpers/bundle-manifest').register(JSON.parse(" +
-    JSON.stringify(JSON.stringify(idToName)) +
-    '));'
-  );
-}
+  let baseUrl =
+    entryBundle.env.outputFormat === 'esmodule' &&
+    entryBundle.env.supports('import-meta-url')
+      ? 'new __parcel__URL__("").toString()' // <-- this isn't ideal. We should use `import.meta.url` directly but it gets replaced currently
```

Comment:
@devongovett Thoughts on this? Seems reasonable that I should be able to use `import.meta.url` without it being replaced? 

I could modify the transform to add a new special parcel symbol (e.g. `__parcel_importMetaUrl__`)? Or potentially add some asset meta data that prevents the stripping?

Reply:
hmm yeah, I guess we could add a new `__parcel` thing, but does this work as is for now? if so I'm ok with leaving it as you have it as well.